src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -378,6 +378,13 @@ public static <K, V extends TopLevelItem> Map<K, V> loadChildren(AbstractFolder<
                     }
                 }
                 name = childNameGenerator.writeItemName(parent, item, effectiveSubdir, childName);
+                String computedName = childNameGenerator.itemNameFromItem(parent, item);
+                if (computedName == null) {
+                    computedName = subdir.getName();
+                }
+                if (!computedName.equals(name)) {
+                    throw new IllegalStateException("Computed name '" + computedName + "' does not match name read from file '" + name + "'");
```

Comment:
```suggestion
                    throw new IllegalStateException("Computed name '" + computedName + "' does not match name written to file '" + name + "'");
```
in this case I guess.